mirrord/layer/src/lib.rs

```diff
@@ -606,21 +606,18 @@ pub(crate) unsafe extern "C" fn close_detour(fd: c_int) -> c_int {
         .get()
         .expect("Should be set during initialization!");
 
-    if SOCKETS.lock().unwrap().remove(&fd).is_some() {
-        FN_CLOSE(fd)
-    } else if *enabled_file_ops
+    let res = FN_CLOSE(fd);
+    if SOCKETS.lock().unwrap().remove(&fd).is_none() && *enabled_file_ops
         && let Some(remote_fd) = OPEN_FILES.lock().unwrap().remove(&fd) {
         let close_file_result = file::ops::close(remote_fd);
 
-        close_file_result
+        let _ = close_file_result
             .map_err(|fail| {
                 error!("Failed closing file with {fail:#?}");
                 -1
```

Comment:
This could be deleted and the map simplified a bit (no block), right?

Reply:
Thanks